src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralConcluder.kt

```diff
@@ -56,7 +58,9 @@ class ReferralConcluder(
   }
 
   private fun deliveryState(referral: Referral): DeliveryState {
-    val approvedActionPlan = referral.approvedActionPlan ?: return DeliveryState.NOT_DELIVERING_YET
+    val approvedActionPlan =
+      actionPlanRepository.findAllByReferralIdAndApprovedAtIsNotNull(referral.id).maxByOrNull { it.approvedAt!! }
```

Comment:
The derived query method (`findAllByReferralIdAndApprovedAtIsNotNull`) could incorporate the `max` query by using `OrderBy` and `Top`: https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repository-query-keywords (perhaps `findTopByOrderByApprovedAtDescByReferralId`?)

However, to simplify this and make more domain-readable, could we instead define `actionPlanRepository.findLatestApprovedActionPlan(referral.id)`?

Reply:
Sure, I couldn't use top or limit in hql but I could use native query:
`  @Query("select * from action_plan where referral_id = :referralId order by approved_at desc limit 1", nativeQuery = true)`  
would this be fine or should we go with your first suggestion `findTopByOrderByApprovedAtDescByReferralId `